packages/utils/src/crypto/randomInteger.ts

```diff
@@ -1,85 +1,70 @@
-//import { randomBytes } from 'crypto'
+import crypto from 'crypto'
 
-// const MAX_SAFE_INTEGER = 2147483648 - WTF????
 /**
- * Returns a random value between `start` and `end`.
- * @example
- * ```
- * randomInteger(3) // result in { 0, 1, 2, 3 }
- * randomInteger(0, 3) // result in { 0, 1, 2, 3 }
- * randomInteger(7, 9) // result in { 7, 8, 9 }
- * randomInteger(8, 8) == 8
- * ```
- * @param start start of the interval
- * @param end end of the interval inclusive
- * @param seed [optional] DO NOT USE THIS set seed manually
- * @returns random number between @param start and @param end
+ * Maximum bit length of a random number.
  */
-export function randomInteger(start: number, end?: number, _seed?: Uint8Array): number {
-  // Our random number generator is broken. FFS FML WTF.
-
-  if (!end) {
-    end = start
-    start = 0
-  }
-  return Math.floor(Math.random() * (end - start)) + start
-  /*
-  if (start < 0 || (end != undefined && end < 0)) {
-    throw Error(`'start' and 'end' must be positive.`)
-  }
-
-  if (end != undefined) {
-    if (start >= end) {
-      throw Error(`Invalid interval. 'end' must be strictly greater than 'start'. Got start: <${start}> end: <${end}>`)
-    }
+const BIT_WIDTH = 64n
 
-    if (start + 1 == end) {
-      return start
-    }
-  } else {
-    if (start == 0) {
-      throw Error(`Cannot pick a random number that is >= 0 and < 0`)
-    }
-  }
+/**
+ * Internal function generating fix length random integer.
+ */
+function nextRandomFullWidth(): bigint {
+  return crypto.randomBytes(Number(BIT_WIDTH / 8n)).readBigUInt64LE()
+}
 
-  // Projects interval from [start, end) to [0, end - start)
-  let interval = end == undefined ? start : end - start
+/**
+ * Maximum random integer that can be generated using randomInteger function.
+ */
+export const MAX_RANDOM_INTEGER = (1n << BIT_WIDTH) - 1n
 
-  if (interval == 1) {
-    return start
-  }
+/**
+ * Internal function generating random integer in half-close interval [0, bound).
+ * Uses an optimized Lemire's method (https://arxiv.org/abs/1805.10941)
+ * as devised in https://github.com/apple/swift/pull/39143 by Stepen Canon.
```

Comment:
Please include the explanation from the GH issue. The issue could be deleted and we'd be missing way too much context.